### DISCUSSION TITLE
Swap postcss for lightningcss

### DISCUSSION THREAD

**Discussion by lukewarlow** [tailwindlabs/tailwindcss#11040] [Ideas]
It would be cool if tailwind was no longer dependent on postcss, a potential alternative is lightning css which should be faster. https://lightningcss.dev/ it has built in prefixing as well so handles the autoprefixer use case too.

**Comment by astratagem** [comment]
> It would be cool if tailwind was no longer dependent on postcss

Why might one prefer Lightning CSS over PostCSS? Aside from its performance claims, which seems to be the basis of that project's branding. Generally a tool written in Rust is likely to be more performant and type-safe than Node.js (assuming eqivalent code quality), so performance in itself doesn't seem like a great selling point.

I neither agree nor disagree, but PostCSS is very established and FWIW this is the first time I've heard of Lightning CSS, so I'd be interested to hear a good argument for this particular Rust-based alternative to PostCSS.

**Comment by adamwathan** [comment]
Already planning on it — it's already integrated in `master` and has been for a few months 🫣

**Reply by lukewarlow** [reply]
Recently found the oxide engine with it, forgot to close this discussion. Great to hear!

**Comment by gyhyfj** [comment]
I'm exected about this  and interested in when'll it be stable?  Cause vite with lightning is almost coming

**Reply by MattCCC** [reply]
Vite already mentions lightningcss in the docs

**Reply by lukewarlow** [reply]
Fwiw I tried to get Vite working with an insider Tailwind build and lightning and I couldn't quite get the hang of it. So I think there's a bit to go first.

**Comment by Nefcanto** [comment]
Anything related to performance is precious. Let's not forget that Tailwind code compiles to 10K while Bootstrap is 70K. One of the major reasons we migrated to Tailwind in the first place was its performance. As we write more and more code, we need to reduce the compilation and bundling time, and file size over the network. We are also very excited to hear that the Tailwind team has a plan for Lightning CSS.

**Comment by tcolinpa** [comment]
Was this launched in v3.4 as said in a previous tailwind blog post about Oxide?

**Reply by lukewarlow** [reply]
No it was pushed back to 3.5 (or later)

**Reply by tcolinpa** [reply]
thanks @lukewarlow!

**Comment by phil-sola** [comment]
I just searched release notes on GH and can't find any mention of Lightning CSS - has this happened already or still not yet?

**Reply by RobinMalfait** [reply]
Tailwind CSS v4 uses Lightning CSS to optimize / minify the output file.

If you are using `@tailwindcss/postcss` then you are still using PostCSS because it's a PostCSS plugin, but the optimization / minification happens via Lightning CSS.

**Reply by Illyism** [reply]
https://nextjs.org/docs/app/api-reference/config/next-config-js/useLightningcss is supported in Next.js, can we integrate this?
